metricbeat/module/logstash/metricset.go

```diff
@@ -15,16 +15,37 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package beat
+package logstash
 
-// Config defines the structure for the Beat module configuration options
-type Config struct {
-	XPackEnabled bool `config:"xpack.enabled"`
+import (
+	"github.com/elastic/beats/v7/metricbeat/helper"
+	"github.com/elastic/beats/v7/metricbeat/helper/elastic"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+// MetricSet can be used to build other metricsets within the Logstash module.
+type MetricSet struct {
+	mb.BaseMetricSet
+	*helper.HTTP
+	XPack bool
 }
 
-// DefaultConfig returns the default configuration for the Beat module
-func DefaultConfig() Config {
-	return Config{
-		XPackEnabled: false,
+// NewMetricSet creates a metricset that can be used to build other metricsets
+// within the Logstash module.
+func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
+	config := elastic.DefaultModeConfig()
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	http, err := helper.NewHTTP(base)
+	if err != nil {
+		return nil, err
 	}
+
+	return &MetricSet{
+		base,
+		http,
+		config.GetMode() == elastic.ModeStackMonitoring,
+	}, nil
 }
```

Comment:
I have no idea why GitHub thinks this is a rename! I deleted the file on the left completely and created the file on the right completely from scrath.